### ISSUE TITLE
Untweetable species Twitterbot thing 🐜 🐌

### ISSUE THREAD

**Issue by hollylkirk** [ropensci/ozunconf18#13]
Some of my colleagues just published a [paper ](https://www.sciencedirect.com/science/article/pii/S1617138118301183?dgcid=coauthor) showing that 57% of Australian threatened species have been mentioned fewer than 20 times on Twitter.  We have named the species that have never been tweeted about the #untweetables.  :frog: :bug: :snake: :beetle:

I propose a project to do the following:
- regularly scrape Twitter to look for mentions of the 40 untweetable species that have never been tweeted about.
- update the original dataset from the paper and do some live plotting
- create a twitterbot that automatically retweets any tweets that mention the 40 zero tweet species and then tweets the updated live plot.

NB. I have 0 experience with doing Twitter stuff in R, but I am very enthusiastic! :smiley_cat:

**Comment by adam-gruer** [comment]
![Noice](https://gph.is/2Bcihop)

Maybe the tweetbot could also tweet pics and info about the #untweetables.  Give them a boost. the earlier dataset could filter out tweetbot generated tweets.

**Comment by hollylkirk** [comment]
Yes! I love this!  Although, online pics are not available/hard to find for many of the #untweetables... But I guess we could pre-generate that content in advance?

**Comment by adam-gruer** [comment]
I don't know if its appropriate, I certainly don't know how doable it is but it could cause buzz interest- raise awareness in a geeky fun way. There is a twitter user who posts "photos" of his 6 year son's drawings as if these creations existed in the real world - he uses some machine learning to generate the pictures.  Maybe a machine learning, natural language processing algorithm to generate fantastical images of the untweetables based on textual descriptions of them or just their latin names? https://www.boredpanda.com/kid-drawings-things-i-have-drawn-dom/?utm_source=google&utm_medium=organic&utm_campaign=organic

**Comment by hollylkirk** [comment]
Wow! That would be so much fun!! Some of these species have really strange common names too (e.g. antbed parrot moth and  robust white-eye) which maybe we could use to generate the image, and then compare it to the real thing...?

**Comment by peggynewman** [comment]
Yep @adam-gruer, this exists, check it out: [Microsoft researchers build a bot that draws what you tell it to](https://blogs.microsoft.com/ai/drawing-ai/) !
Also @hollylkirk, you can use ALA via ALA4R to find:
- most of the common names and synonyms for those species
- images of the species (hopefully)
- lists of threatened species in each Australian jurisdiction
I follow a heap of naturalist pages on Facebook, I wonder if we could apply the same logic to facebook? That's at risk of muddying your awesome "untweetable" title/name.

**Comment by adam-gruer** [comment]
un@ables? I'll see myself out

**Comment by RPanczak** [comment]
Perhaps we could also generate a hex sticker for each animal! ;-)

**Comment by hollylkirk** [comment]
Oh my! 😍

**Comment by RPanczak** [comment]
@hollylkirk authors of the study that you pointed to mention 

> 496 species from multiple taxa

Do you think there is a chance to get a list of them? 

We have 5.6 milion of geolocated 🌐 tweets for Oz 🇦🇺 available (ie since mid May this year) - I could have a quick look 👓 if anything got mentions there? Only few are available in _Table 1_ though.

Btw, could #Rstats ninjas help to advise how to efficiently search for occurrence of string in a column of dataframe?

**Comment by hollylkirk** [comment]
Wow!

@RPanczak I do have access to the full list of threatened species used in the original paper, but I think it would be great to narrow in on the subset that have never been mentioned, or only been mentioned <5 times. 

I will attach the appropriate species list to my next comment... 🔢

**Comment by peggynewman** [comment]
Potentially useful active twitter accounts for testing: @TSR_Hub (Threatened Species Recovery Hub) and @TSCommissioner (Threatened Species Commissioner)

**Comment by smwindecker** [comment]
cool @hollylkirk! I wonder if instead of a static untweetables list, the list could update as tweets change their rankings?
